controller.py

```diff
@@ -218,14 +218,65 @@ def start(self, _) -> None:
         
         for enemy in self.enemies:
             enemy.start_move()
-        
 
     def on_game_end(self) -> None:
-        """##Fonction appelée lorsque la partie est terminée afin d'afficher le menu de score et de sauvegarder le score """
+        """##Fonction appelée lorsque la partie est terminée afin d'afficher le menu de score et de sauvegarder le
+        score """
         self.frame.destroy()
-        print("You died")
-        print(f"Your score: {self.player.score.value}")
-        #self.root.score_controller.start()
+        self.root.controller = GameEndController(self.root, self.player.score.value)
+
+
+class GameEndController(Controller):
+    """#Controlleur de fin de partie
+    
+    Cette classe gère la fin de partie. Elle est une sous-classe de Controller. Sa responsabilité est de gérer les événements
+    de fin de partie ainsi que les interactions entre les différents objets de fin de partie.
+    Tel:
+    - Afficher le menu de fin de partie
+    - Sauvegarder le score, si le joueur entre un nom et appuie sur entrer.
+        - Afficher le tableau des highscores
+    - Afficher le menu principal, si le boutton est pressé.
+    - Afficher le menu des options, si le boutton est pressé.
+
+    Attributs:
+        - view (GameEndView): La vue de fin de partie
+        - root (Root): La fenêtre principale du jeu
+        - frame (tk.Frame): Le frame dans lequel le controlleur est affiché
+        - score (int): Le score du joueur
+    """
+    def __init__(self, root: Root, score: int):
+        """"""
+        super().__init__(root, tk.Frame(root))
+        self.score = score
+        self.view = GameEndView(root, score)
+        self.root.title("Game Over")
+        self.view.draw()
+        self.initialize()
+
+    def initialize(self) -> None:
+        """##Fonction appelée pour initialiser le controlleur de fin de partie
+
+        Cette fonction initialise les événements du controlleur de fin de partie
+        """
+        # Event listener to get back the string from nameEntry inside the view
+        self.view.nameEntry.bind("<Return>", self.on_submit)
+        self.view.nameEntry.focus_set()
+        self.view.btn_menu.bind("<Button-1>", self.on_menu)
+
+    def on_submit(self, _) -> None:
+        """##Fonction appelée lorsque le joueur appuie sur Entrée pour valider son nom"""
+        name = self.view.nameEntry.get()  # Get the name from the entry
+        if name != "":
```

Comment:
```suggestion
        if name:
```
Comparaison booléene avec un string donne True seulement si elle n'est pas vide